modules/highgui/src/window_gtk.cpp

```diff
@@ -1973,7 +1982,25 @@ static gboolean icvOnMouse( GtkWidget *widget, GdkEvent *event, gpointer user_da
 //        if((unsigned)pt.x < (unsigned)(image_widget->original_image->width) &&
 //           (unsigned)pt.y < (unsigned)(image_widget->original_image->height) )
         {
-            int flags = (state & GDK_SHIFT_MASK ? CV_EVENT_FLAG_SHIFTKEY : 0) |
+            int flags = 0;
+            if (state & GDK_SCROLL_MASK) {
+                cv_event = CV_EVENT_MOUSEMOVE;
+                int orient = CV_EVENT_MOUSEWHEEL;
+                switch(event->scroll.direction) {
+                case GDK_SCROLL_RIGHT: orient = CV_EVENT_MOUSEHWHEEL;
+                case GDK_SCROLL_UP:    flags |= ((1 << 16) | orient);
+                    break;
+                case GDK_SCROLL_LEFT:  orient = CV_EVENT_MOUSEHWHEEL;
+                case GDK_SCROLL_DOWN:  flags |= ((-1 << 16) | orient);
```

Comment:
Nice question.. Well, the structure GdkEventScroll (event->scroll) contains delta_x and delta_y, which can be received from gdk-event-get-scroll-deltas call.
https://developer.gnome.org/gdk3/3.4/gdk3-Event-Structures.html#GdkEventScroll
I'll play with deltas and improve the code.

_I suppose, in the future CvMouseCallback can be remastered to easier access to the data of mouse event, e.g by replacing arguments 'int event, int x, int y, int flags' to 'const CvMouseEvent *'._